### PR TITLE
Finally, turn the kernel into a lib !

### PR DESCRIPTION
**Pull request by n1tram1**
The "kernel" is now a library which boards (separate crates) can use.

Co-authored-by: Esteban Blanc <estblcsk@gmail.com>

### CODE REVIEW

File: aarch64_qemuvirt/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,77 @@
+#![no_std]
+#![no_main]
+// #![feature(custom_test_frameworks)]
+// #![test_runner(crate::kernel_tests::runner)]
+// #![reexport_test_harness_main = "ktests_launch"]
+
+#[cfg(not(target_arch = "aarch64"))]
+compile_error!("Must be compiled as aarch64");
```
Comment (Skallwar) [inline aarch64_qemuvirt/src/main.rs:8]: Is this still necesarry ?

File: aarch64_qemuvirt/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+name = "aarch64_qemuvirt"
+version = "0.1.0"
+edition = "2021"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[dependencies]
+kernel = { path = "../kernel", features = ["aarch64", "aarch64_pgt48oa"] }
+
+tock-registers = "0.7.x"
```
Comment (Skallwar) [inline aarch64_qemuvirt/Cargo.toml:11]: This should be in the kernel `Cargo.toml`, right?

File: riscv64_qemuvirt/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,65 @@
+#![no_std]
+#![no_main]
+#![feature(custom_test_frameworks)]
+#![test_runner(kernel::super::kernel_tests::runner)]
```
Comment (Skallwar) [inline riscv64_qemuvirt/src/main.rs:4]: Is this not broken?

File: aarch64_qemuvirt/src/main.rs (hunk 1 above)
Comment (n1tram1) [reply aarch64_qemuvirt/src/main.rs:8]: Yes, it's a safety check, I think it is better than a cryptic compiler error in the future

File: riscv64_qemuvirt/src/main.rs (hunk 3 above)
Comment (n1tram1) [reply riscv64_qemuvirt/src/main.rs:4]: Yes

**Comment by Skallwar** [comment]
What should we do about the CI?

**Comment by n1tram1** [comment]
> What should we do about the CI?

Fixed

**Pull request merged**